### PR TITLE
support media lists

### PR DESCRIPTION
**Pull request by takeit**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | no
| New feature?              | yes
| BC breaks?                | no
| Deprecations?             | no
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | SWP-1114
| License                   | AGPLv3

### CODE REVIEW

File: features/content_push/send_content_with_custom_media_fields.feature [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  Scenario: Saving the data from custom media fields
+    Given I am authenticated as "test.user"
+    When I add "Content-Type" header equal to "application/json"
+    And I send a "POST" request to "/api/v1/assets/push" with parameters:
+      | key          | value                 |
+      | media_id     | 1234567890987654321a  |
+      | media        | @image.jpg            |
+    Then the response status code should be 201
+
+    And I am authenticated as "test.user"
```
Comment (ahilles107) [inline features/content_push/send_content_with_custom_media_fields.feature]: You don't need to be authenticted to use this endpoint. Instead that you can add header with security hash

File: src/SWP/Bundle/BridgeBundle/Resources/config/serializer/Model.Package.yml [hunk 2]
```diff
@@ -15,3 +15,6 @@ SWP\Component\Bridge\Model\Package:
         updatedAt:
             expose: true
             type: DateTime
+        groups:
+            expose: true
+            type: ArrayCollection<string,SWP\Component\Bridge\Model\Group>
```
Comment (ahilles107) [inline src/SWP/Bundle/BridgeBundle/Resources/config/serializer/Model.Package.yml]: new line at end of file missing

File: src/SWP/Bundle/ContentBundle/EventListener/ProcessArticleMediaListener.php [hunk 3]
```diff
@@ -63,10 +64,29 @@ public function onArticleCreate(ArticleEvent $event)
         $article = $event->getArticle();
 
         if (null === $package || (null !== $package && 0 === count($package->getItems()))) {
-            return;
+            //return;
```
Comment (ahilles107) [inline src/SWP/Bundle/ContentBundle/EventListener/ProcessArticleMediaListener.php]: remove it if not needed

Comment (takeit) [reply src/SWP/Bundle/ContentBundle/EventListener/ProcessArticleMediaListener.php]: still wip

File: src/SWP/Bundle/BridgeBundle/Resources/config/doctrine-orm/Group.orm.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+SWP\Component\Bridge\Model\Group:
```
Comment (ahilles107) [inline src/SWP/Bundle/BridgeBundle/Resources/config/doctrine-orm/Group.orm.yml]: we should add indexes here. One for (package, deletedAt). Not sure about code? Its used in query filters?

File: src/SWP/Bundle/BridgeBundle/Resources/config/doctrine-orm/Group.orm.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            generator:
+                strategy: AUTO
+    fields:
+        code:
+            type: string
+        deletedAt:
+            type: datetime
+            nullable: true
+    oneToMany:
+        items:
```
Comment (ahilles107) [inline src/SWP/Bundle/BridgeBundle/Resources/config/doctrine-orm/Group.orm.yml]: We will have issues with it. This way it's always lazy loaded when group is loaded (if not already joined in query). We have problem like that with content list items. Imo we should remove it and work with items through service.

File: src/SWP/Bundle/BridgeBundle/Serializer/PackageSubscriber.php [hunk 6]
```diff
@@ -51,12 +51,29 @@ public function onPostDeserialize(ObjectEvent $event)
...
                 $item->setName($key);
             }
         }
     }
 
+    private function processGroups(PackageInterface $package): void
+    {
+        foreach ((array) $package->getGroups() as $groups) {
+            foreach ((array) $groups as $group) {
+                foreach ($group->getItems() as $groupItem) {
```
Comment (ahilles107) [inline src/SWP/Bundle/BridgeBundle/Serializer/PackageSubscriber.php]: we should get group items through service

**Pull request merged**